File: all_banchmark/02_prime_sieve/prime_sieve.c
// Prime Sieve Benchmark - C
// Sieve of Eratosthenes to test array access and bit operations

#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>

// Trial division to match Knoz implementation
int main() {
    int64_t limit = 100000;  // 100K to match Knoz
    int64_t count = 0;
    
    for (int64_t n = 2; n <= limit; n++) {
        int is_prime = 1;
        for (int64_t d = 2; d * d <= n; d++) {
            if (n % d == 0) {
                is_prime = 0;
                break;
            }
        }
        if (is_prime) count++;
    }
    
    printf("Primes up to %lld: %lld\n", limit, count);
    return 0;
}

File: all_banchmark/02_prime_sieve/prime_sieve.rs
// Prime Sieve Benchmark - Rust
// Sieve of Eratosthenes to test array access and bit operations

// Trial division to match Knoz implementation
fn main() {
    let limit: i64 = 100_000;  // 100K to match Knoz
    let mut count: i64 = 0;
    
    let mut n: i64 = 2;
    while n <= limit {
        let mut is_prime = true;
        let mut d: i64 = 2;
        while d * d <= n {
            if n % d == 0 {
                is_prime = false;
                break;
            }
            d += 1;
        }
        if is_prime {
            count += 1;
        }
        n += 1;
    }
    
    println!("Primes up to {}: {}", limit, count);
}
